### DISCUSSION TITLE
SHA1 Fingerprint being displayed

### DISCUSSION THREAD

**Discussion by iyyapa** [openssl/openssl#24893] [Q&A] [answered]
When we run CLI “show crypto ca certificates ” , it is showing SHA1 fingerprint.

b143-BL1# show crypto ca certificates

_Certificate Type: https; Trustpoint: none
certificate:
SHA1 Fingerprint=84:20:18:12:C9:50:3C:CE:DE:B5:3B:94:6A:33:19:5A:4E:00:36:8E_

Can someone clarify by default SHA1 fingerprint is default option. When I check in the OPSSL and see that  the cert signature is Encrypted with SHA256 and finger print is not part of certification. 

_[root@B06_L03-111]# openssl x509 -in /etc/fabos/certs/sw0/https_swcert.pem -noout -fingerprint
WARNING: can't open config file: /etc/fabos/openssl.cnf
SHA1 Fingerprint=87:F2:4B:D6:B0:36:AD:A4:9D:91:9D:44:63:00:C9:78:DE:E2:3B:9B

[root@B06_L03-111]# openssl x509 -in /etc/fabos/certs/sw0/https_swcert.pem -noout -fingerprint -sha256
WARNING: can't open config file: /etc/fabos/openssl.cnf
SHA256 Fingerprint=12:14:EE:34:85:C8:67:E5:6F:F9:2F:6F:27:9E:EA:B1:F0:4D:D8:EA:3E:87:0B:6F:17:18:E0:15:B3:78:D9:DE_

Can you confirm, SHA1 is default option for SHA1 fingerprint. We can make SHA256 being displayed as fingerprint instead of SHA1 ?

**Comment by t8m** [comment, marked as answer]
The default digest used for the fingerprint in the x509 command is SHA1. You can change it to SHA256 by adding -sha256 option as you did above.

**Comment by iyyapa** [comment]
openssl x509 -text -in /etc/fabos/certs/sw0/https_swcert.pem -fingerprint -sha256 -- works..
